lib/profileAgents.js

```diff
@@ -602,6 +609,104 @@ async function _updateMeterController({meterId, controller}) {
   }
 }
 
+async function _getProfileAgentRecords({accountId, includeSecrets}) {
+  const query = {'profileAgent.account': accountId};
+  const projection = {_id: 0};
+  const collection = getCollection('profile-profileAgent');
+  let records = await collection.find(query, {projection}).toArray();
+  const zcapRefreshed = await _refreshProfileAgentZcaps({
+    profileAgentRecords: records
+  });
+  if(zcapRefreshed) {
+    records = await collection.find(query, {projection}).toArray();
+  }
+  if(!includeSecrets) {
+    records = await removeSecretsFromRecords({records});
+  }
+  return records;
+}
+
+async function _refreshProfileAgentZcaps({profileAgentRecords}) {
+  let zcapRefreshed = false;
+  const records = klona(profileAgentRecords);
```

Comment:
Do we need to copy -- what's the advantage in this case? Can we modify in place since this is a local helper that is already updating its own copy of records? Just because we moved this to a helper doesn't mean we should have to copy, IMO.

Reply:
No longer making copy. Removed.